dataframely/columns/_base.py

```diff
@@ -97,7 +100,25 @@ def validation_rules(self, expr: pl.Expr) -> dict[str, pl.Expr]:
         if not self.nullable:
             result["nullability"] = expr.is_not_null()
         if self.check is not None:
-            result["check"] = self.check(expr)
+            if isinstance(self.check, dict):
+                for k, v in self.check.items():
+                    result[k] = v(expr)
+
+            elif isinstance(self.check, list):
+                for num, i in enumerate(self.check):
+                    callable_name = i.__name__
+                    if callable_name == "<lambda>":
+                        callable_name = f"check__{num}"
+                    result[callable_name] = i(expr)
+
+            elif callable(self.check):
```

Comment:
Given that we expect the user to leverage `mypy`, we could go into the `else` block here. Otherwise, we'd need to add an `else` block raising an exception (and exclude that from coverage which isn't great IMO).